DockerfileUBI

```diff
@@ -62,11 +72,16 @@ RUN dnf install -y \
 # Copy source tree
 COPY . /build
 
-# Build Monocle
+# Build Monocle with architecture-specific settings
 RUN cd /build && \
     cabal update && \
     cabal build --project-file=cabal-override.project && \
-    cp $(find dist-newstyle/build/*/*/*/*/monocle/build/ -name monocle -type f) /
+    ARCH=$(uname -m) && \
+    if [ "$ARCH" = "x86_64" ] || [ "$ARCH" = "aarch64" ]; then \
+    cp $(find dist-newstyle/build/$ARCH*/*/*/monocle/build/ -name monocle -type f) /; \
+    else \
+    echo "Unsupported architecture: $ARCH" && exit 1; \
+    fi
```

Comment:
We could replace the build and cp command with:

```
cabal install --project-file=cabal-override.project --installdir /
```
That way we wouldn't need the arch to copy the binary.